### PR TITLE
(Towards 2205, Closes 2315) Programs now handle contains statements to result in code blocks

### PR DESCRIPTION
**Pull request by LonelyCat124**
Not sure if this needs adding for _program_handler as well as _main_program_handler or not as I'm not clear on the difference.

This is again a workaround similar to that in place for functions and subroutines - it results in a codeblock but I think this is better than the current disappearing act. Future fixes for #2205 will also need to work for Program statements.

### CODE REVIEW

File: src/psyclone/psyir/frontend/fparser2.py
Comment (arporter) [inline src/psyclone/psyir/frontend/fparser2.py:4586]: Please could you do away with the `has_contains` and just move this raise directly into L4582. L4594 would then just be `pass` (please also add a comment along the lines of "The routine does not have a Contains section").

File: src/psyclone/tests/psyir/frontend/fparser2_main_program_handler_test.py
Comment (arporter) [inline src/psyclone/tests/psyir/frontend/fparser2_main_program_handler_test.py:112]: I would have expect the whole Program to be in the CodeBlock? Does re-generating the code (use the `fortran_writer` fixture) give the correct output?

**Review by arporter — CHANGES_REQUESTED** [review]
Thanks for this @LonelyCat124. Everything that helps towards a successful pass-through is very valuable!

I think the change itself could be tidied slightly and I'm a bit worried that the CodeBlock isn't what I expected. I'm therefore requesting that you extend the test just to make sure.
Apart from that this is a small bug fix so no need to update docs.
I'll check the integration tests next time (but don't anticipate any problems).

File: src/psyclone/psyir/frontend/fparser2.py
Comment (LonelyCat124) [reply src/psyclone/psyir/frontend/fparser2.py:4586]: Done, also fixed typo in the exception.

File: src/psyclone/tests/psyir/frontend/fparser2_main_program_handler_test.py
Comment (LonelyCat124) [reply src/psyclone/tests/psyir/frontend/fparser2_main_program_handler_test.py:112]: Yeah - I should probably check more here, my main thing was to check the subroutine wasn't lost but maybe just checking the output from fortran_writer would also be a good idea.

**Comment by LonelyCat124** [comment]
Should have fixed those issues.

**Review by arporter — APPROVED** [review]
All requested changes have been made (thanks).
Just waiting on the integration tests now before I proceed to merge.

**Comment by arporter** [comment]
Integration tests were green. Will proceed to merge.

**Pull request merged**
